src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_quantization.cpp

```diff
@@ -288,10 +288,16 @@ void prepare_quantization::prepare_scale_shift_opt(program &p, quantize_node& qu
     p.add_connection(in_shift_node, new_quantize_node);
     p.add_connection(out_scale_node, new_quantize_node);
     p.add_connection(out_shift_node, new_quantize_node);
-    new_quantize_node.add_memory_dependency(in_scale_node.get_unique_id());
-    new_quantize_node.add_memory_dependency(in_shift_node.get_unique_id());
-    new_quantize_node.add_memory_dependency(out_scale_node.get_unique_id());
-    new_quantize_node.add_memory_dependency(out_shift_node.get_unique_id());
+    if (new_quantize_node.likely_from_mempool()) {
+        if (in_scale_node.likely_from_mempool())
+            new_quantize_node.add_memory_dependency(in_scale_node.get_unique_id());
```

Comment:
How about modify add_memory_dependency to check the likeliness inside the function, instead of checking before add_memdep every time?

Reply:
done